pandas/io/formats/style.py

```diff
@@ -2423,6 +2440,50 @@ def set_table_styles(
             self.table_styles = table_styles
         return self
 
+    def set_descriptors(
+        self, descriptors: list[Descriptor | tuple[str, Descriptor]] | None = None
+    ) -> Styler:
+        """
+        Add header-level calculations to the output which describes the data.
+
+        .. versionadded:: 1.5.0
+
+        Parameters
+        ----------
+        descriptors : list of str, callables or 2-tuples of str and callable
+            If a string is given must be a valid Series method, e.g. "mean" invokes
```

Comment:
Could you some additional details about the callable.  Specifically it must (or should) return a scalar quantity.  Is the callable allowed to throw and exception?  If so, what happens?  I would be good to clarify this.